src/select2.component.ts

```diff
@@ -148,6 +148,29 @@ export class Select2Component implements AfterViewInit, OnChanges, OnDestroy, On
             this.renderer.setElementProperty(this.selector.nativeElement, 'disabled', this.disabled);
         }
     }
+     
+    private setElementValue ( newValue ) {
+        // NOTE:
+        // Multiple values are received from ngOnChanges as a string with single values separated by comma.
+        // Select2OptionData id values should not contain commas for predictable results.
+        // While using commas in ids is a bad idea generally, here may cause malfunctions.
+
+        if (newValue.indexOf(',') != -1) {
+            var singleVals = newValue.split(',');
```

Comment:
I'm not sure if with "users" you're referring to developers using the library, but the (valueChanged) event is passing comma separated strings. I think it's better to stick with a format, anyway, my latest version accepts both strings and arrays.